src/isolate.rs

```diff
@@ -45,6 +46,13 @@ impl Isolate {
             IsolateHolder::Borrowed(p) => p,
         }
     }
+
+    pub fn current_context(&self) -> context::Context {
+        unsafe { 
```

Comment:
I guess that'd be the 3rd option. It works out well and prevents the crash (and it did crash).